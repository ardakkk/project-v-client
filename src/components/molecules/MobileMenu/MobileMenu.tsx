import * as React from 'react';

import Badge from '@material-ui/core/Badge';
import IconButton from '@material-ui/core/IconButton';
import Menu from '@material-ui/core/Menu';
import MenuItem from '@material-ui/core/MenuItem';
import Brightness5Icon from '@material-ui/icons/Brightness5';
import NotificationsIcon from '@material-ui/icons/Notifications';
import SettingsIcon from '@material-ui/icons/Settings';

interface IMobileMenuProps {
    mobileMoreAnchorEl?: any;
    handleMobileMenuClose?: () => void;
}

export const MobileMenu: React.FunctionComponent<IMobileMenuProps> = ({
    mobileMoreAnchorEl,
    handleMobileMenuClose,
}) => {
    const isMobuleMenuOpenMenu: boolean = Boolean(mobileMoreAnchorEl);

    return (
        <Menu
            anchorEl={mobileMoreAnchorEl}
            anchorOrigin={{ vertical: 'top', horizontal: 'right' }}
            keepMounted={true}
            transformOrigin={{ vertical: 'top', horizontal: 'right' }}
            open={isMobuleMenuOpenMenu}
            onClose={handleMobileMenuClose}>
            <MenuItem>
                <IconButton color='inherit'>
                    <Brightness5Icon />
                </IconButton>
                <p>white-theme</p>
            </MenuItem>
            <MenuItem>
                <IconButton
                    aria-label='show 11 new notifications'
                    color='inherit'>
                    <Badge badgeContent={11} color='secondary'>
                        <NotificationsIcon />
                    </Badge>
                </IconButton>
                <p>Notifications</p>
            </MenuItem>
            <MenuItem>
                <IconButton
                    aria-label='account of current user'
                    aria-controls='primary-search-account-menu'
                    aria-haspopup='true'
                    color='inherit'>
                    <SettingsIcon />
                </IconButton>
                <p>Profile</p>
            </MenuItem>
        </Menu>
    );
};
