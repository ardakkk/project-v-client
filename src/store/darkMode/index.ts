import { Reducer } from 'redux';

import { updateObject } from '../../services/shared/utility';
import { DarkModeTypes, IDarkModeState } from './types';

const INITAL_STATE: IDarkModeState = {
    isDarkMode: false,
};

const reducer: Reducer<IDarkModeState> = (state = INITAL_STATE, action) => {
    switch (action.type) {
        case DarkModeTypes.SET_DARK_MODE:
            return updateObject(state, {
                isDarkMode: action.isDarkMode,
            }) as IDarkModeState;
        default:
            return state;
    }
};

export default reducer;
