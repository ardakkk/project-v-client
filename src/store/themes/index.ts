import { Reducer } from 'redux';

import { DarkModeTypes, IDarkModeState } from './types';

const INITAL_STATE: IDarkModeState = {
    theme: '',
};

const reducer: Reducer<IDarkModeState> = (state = INITAL_STATE, action) => {
    switch (action.type) {
        case DarkModeTypes.SET_DARK_MODE:
            return { ...state, theme: action.payload.theme };
        default:
            return state;
    }
};

export default reducer;
