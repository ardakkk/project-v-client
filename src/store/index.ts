import { applyMiddleware, compose, createStore, Store } from 'redux';
import createSagaMiddleware from 'redux-saga';

import { IDarkModeState } from './darkMode/types';
import rootReducer from './rootReducer';

export interface IApplicationState {
    darkMode: IDarkModeState;
}

const composeEnhancers =
    process.env.NODE_ENV === 'development'
        ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
        : null || compose;
const sagaMiddleware = createSagaMiddleware();
const store: Store<IApplicationState> = createStore(
    rootReducer,
    composeEnhancers(applyMiddleware(sagaMiddleware))
);

export default store;
