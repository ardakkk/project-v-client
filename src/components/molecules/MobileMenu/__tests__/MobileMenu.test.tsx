import { shallow } from 'enzyme';
import * as React from 'react';
import { Provider } from 'react-redux';

import store from '../../../../store';
import MobileMenu from '../MobileMenu';

describe('MobileMenu component.', () => {
    it('Snapshot should match.', () => {
        const component = shallow(
            <Provider store={store}>
                <MobileMenu />
            </Provider>
        );

        expect(component).toMatchSnapshot();
    });
});
