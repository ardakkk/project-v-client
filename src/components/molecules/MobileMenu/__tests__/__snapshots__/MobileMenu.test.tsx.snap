// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileMenu component. Snapshot should match. 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Connect(MobileMenu) />
</ContextProvider>
`;
